include/toml++/impl/path.h

```diff
@@ -6,42 +6,69 @@
 
 #include "forward_declarations.h"
 #include "std_vector.h"
-#include "std_variant.h"
 #include "header_start.h"
 
 TOML_NAMESPACE_START
 {
 	/// \brief Indicates type of path component, either a key, an index in an array, or invalid
 	enum class TOML_CLOSED_ENUM path_component_type : uint8_t
 	{
-		invalid		= 0x0,
 		key			= 0x1,
 		array_index = 0x2
 	};
 
-	/// \brief Holds the value of a path component, either the name of the key in a string_view, or the index of an array as a size_t
-	using path_component_value = std::variant<std::string, size_t>;
-
 	/// \brief Represents a single component of a complete 'TOML-path': either a key or an array index
 	struct TOML_EXPORTED_CLASS path_component
 	{
 		friend class path;
```

Comment:
```suggestion

```

Probably doesn't need to be a friend of `path` at all now that all the state handling has moved to the `path_component` itself